internal/docker/podman.go

```diff
@@ -0,0 +1,236 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/anchore/stereoscope/internal/log"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	// Podman supports the following connections
+	// rootless "unix://run/user/$UID/podman/podman.sock" (Default)
+	// rootfull "unix://run/podman/podman.sock (Default)
+	// remote rootless ssh://engineering.lab.company.com/run/user/$UID/podman/podman.sock
+	// remote rootfull ssh://root@10.10.1.136:22/run/podman/podman.sock
+
+	localRootlessPath     = getLocalRootlesPath()
+	defaultRemoteRootless = fmt.Sprintf("ssh://core@localhost:63753%s?secure=false", localRootlessPath)
+	defaultLocalRootless  = fmt.Sprintf("unix://%s", localRootlessPath)
+)
+
+func getLocalRootlesPath() string {
+	return fmt.Sprintf("unix:///run/user/%d/podman/podman.sock", os.Getuid())
+}
+
+func podmanOverSSH() (*client.Client, error) {
+	log.Debug("using docker client to connect to podman over ssh")
+	var clientOpts = []client.Opt{
+		client.WithAPIVersionNegotiation(),
+		// TODO: is this needed?
+		client.FromEnv,
```

Comment:
will this mess up settings based on `DOCKER_HOST`? (should we not respond to docker-specific args?)